File: twitter-fetcher/src/test/java/org/jesperancinha/twitter/model/UserTest.java
package org.jesperancinha.twitter.model;

import com.google.gson.FieldNamingPolicy;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import org.jesperancinha.twitter.model.twitter.User;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class UserTest {

    private final static Gson gson = new GsonBuilder()
            .setDateFormat("EEE MMM dd HH:mm:ss ZZZZZ yyyy")
            .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
            .create();


    @Test
    void testEquals_whenEquals_Ok() {
        final User user1 = gson.fromJson(
                "{\n" +
                        "      \"id\": 988105075267637248,\n" +
                        "      \"id_str\": \"988105075267637248\",\n" +
                        "      \"name\": \"Climate Alliance Switzerland\",\n" +
                        "      \"screen_name\": \"climalliancech\",\n" +
                        "      \"created_at\": \"Sun Apr 22 17:19:42 +0000 2018\"\n" +
                        "    }", User.class);

        final User user2 = gson.fromJson(
                "{\n" +
                        "      \"id\": 988105075267637248,\n" +
                        "      \"id_str\": \"988105075267637248\",\n" +
                        "      \"name\": \"Climate BALALLA Switzerland\",\n" +
                        "      \"screen_name\": \"UUUU\",\n" +
                        "      \"created_at\": \"Sun Jan 22 17:19:42 +0000 2018\"\n" +
                        "    }", User.class);

        assertThat(user1).isEqualTo(user2);
    }

    @Test
    void testHashCode_whenOther_NotSame() {
        final User user1 = gson.fromJson(
                "{\n" +
                        "      \"id\": 988105075267637248,\n" +
                        "      \"id_str\": \"988105075267637248\",\n" +
                        "      \"name\": \"Climate Alliance Switzerland\",\n" +
                        "      \"screen_name\": \"climalliancech\",\n" +
                        "      \"created_at\": \"Sun Apr 22 17:19:42 +0000 2018\"\n" +
                        "    }", User.class);

        final User user2 = gson.fromJson(
                "{\n" +
                        "      \"id\": 988105075267637248,\n" +
                        "      \"id_str\": \"988105075267637248\",\n" +
                        "      \"name\": \"Climate BALALLA Switzerland\",\n" +
                        "      \"screen_name\": \"UUUU\",\n" +
                        "      \"created_at\": \"Sun Jan 22 17:19:42 +0000 2018\"\n" +
                        "    }", User.class);

        assertThat(user1).isNotSameAs(user2);
        assertThat(user1.hashCode()).isEqualTo(user2.hashCode());
    }
}

File: news-cast-explorer-fetcher/src/test/kotlin/org/jesperancinha/newscast/model/source/UserTest.kt
package org.jesperancinha.newscast.model.source

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import io.kotest.matchers.shouldBe
import io.kotest.matchers.types.shouldNotBeSameInstanceAs
import org.junit.jupiter.api.Test

internal class UserTest {
    private val objectMapper = ObjectMapper()

    @Test
    @Throws(JsonProcessingException::class)
    fun testEquals_whenEquals_Ok() {
        val user1 = objectMapper.readValue(
            """{
      "id": 988105075267637248,
      "id_str": "988105075267637248",
      "name": "Climate Alliance Switzerland",
      "screen_name": "climalliancech",
      "created_at": "Sun Apr 22 17:19:42 +0000 2018"
    }""", User::class.java)
        val user2 = objectMapper.readValue(
            """{
      "id": 988105075267637248,
      "id_str": "988105075267637248",
      "name": "Climate BALALLA Switzerland",
      "screen_name": "UUUU",
      "created_at": "Sun Jan 22 17:19:42 +0000 2018"
    }""", User::class.java)
        user1 shouldBe user2
    }

    @Test
    @Throws(JsonProcessingException::class)
    fun testHashCode_whenOther_NotSame() {
        val user1 = objectMapper.readValue(
            """{
      "id": 988105075267637248,
      "id_str": "988105075267637248",
      "name": "Climate Alliance Switzerland",
      "screen_name": "climalliancech",
      "created_at": "Sun Apr 22 17:19:42 +0000 2018"
    }""", User::class.java)
        val user2 = objectMapper.readValue(
            """{
      "id": 988105075267637248,
      "id_str": "988105075267637248",
      "name": "Climate BALALLA Switzerland",
      "screen_name": "UUUU",
      "created_at": "Sun Jan 22 17:19:42 +0000 2018"
    }""", User::class.java)
        user1 shouldNotBeSameInstanceAs user2
        user1.hashCode() shouldBe user2.hashCode()
    }
}